packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -227,6 +238,48 @@ const App = (props) => {
         })
     }, [])
 
+    const {usid} = useUsid()
+    const createShopperContext = useShopperContextsMutation('createShopperContext')
+    const deleteShopperContext = useShopperContextsMutation('deleteShopperContext')
+    const updateShopperContext = useShopperContextsMutation('updateShopperContext')
+    const {data: shopperContext} = useShopperContext(
+        {
+            parameters: {usid: usid, siteId: site.id}
+        },
+        {
+            enabled: !isServer,
+            onError: async () => {
+                await createShopperContext.mutateAsync({
+                    parameters: {usid, siteId: site.id},
+                    body: {}
+                })
+            }
+        }
+    )
+    console.log('shopperContext', shopperContext)
+    const updateShopperContextObj = useShopperContextSearchParams()
+
+    useEffect(() => {
+        const executeShopperContextUpdate = async () => {
+            // update the shopper context if the query string contains the relevant search parameters
+            console.log('updating shoppercontext', updateShopperContextObj)
+            await updateShopperContext.mutateAsync({
+                parameters: {usid, siteId: site.id},
+                body: updateShopperContextObj
+            })
+            // Refresh to update the data on the page
+            refetchDataOnClient()
+        }
+        executeShopperContextUpdate()
+    }, [location.search])
```

Comment:
Can we encapsulate these logic inside a custom hook instead of having all of them in the app component? 
```
// needs a better naming
const {currentContext} = usePwaKitShopperContext()
```